wixoaGit/OpenDream
# Partially automate unit tests

How to create a unit test with the new system:

1. Create the DM file for the test somewhere in `Content.Tests/DMProject/Tests`
2. Include the file in the DME. In the future we should just auto-generate the list of files to compile but I couldn't be bothered to set that up.
3. If the test is supposed to runtime, put `// RUNTIME TRUE` on the first line. The only whitespace that matters is having a space between `RUNTIME` and `TRUE`
4. Declare the test's proc to call like so: `/world/proc/[filename]_Proc()` without the file extension.
5. Voila

I didn't move every single existing test over because I didn't want to rewrite all of them, but this makes it easier to add more tests.
